Azure/azure-cli
# [security] Add security assessment commands

**Description**  


Adding assessment management cmdlets to az security module
- assessment - list, show, create, delete
- assessment-metadata - list, show, create, delete
- sub-assessment - list, show

**Testing Guide**  
Assessments and sub assessments are generated by Microsoft.Security RP if there are any relevant insights on your subscription.
AssessmentMetadata shows all the available assessment types and by default returns the built in assessment types.

Users can create their own managed assessment by creating a metadata first:
   `az security assessment-metadata create -n "8f211ea8-777b-45c4-8631-776d535afd62" --display-name "test assessment" --severity "High" --description "hello world"`
and then create assessments on specific resources:
   `az security assessment create -n 8f211ea8-777b-45c4-8631-776d535afd62 --status-code Healthy`
When a metadata is deleted, all the assessments from this type is deleted as well

---

This checklist is used to make sure that common guidelines for a pull request are followed.
